VocaDb.ResXFileCodeGenerator/FileOptions.cs

```diff
@@ -0,0 +1,114 @@
+﻿using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+namespace VocaDb.ResXFileCodeGenerator;
+
+public sealed record FileOptions
+{
+	public string InnerClassInstanceName { get; init; }
+	public string InnerClassName { get; init; }
+	public InnerClassVisibility InnerClassVisibility { get; init; }
+	public bool PartialClass { get; init; }
+	public bool StaticMembers { get; init; } = true;
+	public AdditionalText File { get; init; }
+	public string FilePath { get; init; }
+	public bool StaticClass { get; init; }
+	public bool NullForgivingOperators { get; init; }
+	public bool PublicClass { get; init; }
+	public string ClassName { get; init; }
+	public string? CustomToolNamespace { get; init; }
+	public string LocalNamespace { get; init; }
+	public bool Valid { get; init; }
+
+	public FileOptions()
+	{
+		LocalNamespace = "";
+		CustomToolNamespace = "";
+		ClassName = "";
+		InnerClassInstanceName = "";
+		InnerClassName = "";
+		File = null!;
+		FilePath = "";
+	}
+
+	private FileOptions(AdditionalText file, AnalyzerConfigOptions options, GlobalOptions globalOptions)
+	{
+		File = file;
+		var resxFilePath = file.Path;
+		LocalNamespace = Utilities.GetLocalNamespace(
+			resxFilePath,
+			options.TryGetValue("build_metadata.EmbeddedResource.TargetPath", out var targetPath) &&
+			targetPath is { Length: > 0 }
+				? targetPath
+				: null,
+			globalOptions.ProjectFullPath,
+			globalOptions.RootNamespace);
+
+		CustomToolNamespace =
+			options.TryGetValue("build_metadata.EmbeddedResource.CustomToolNamespace",
+				out var customToolNamespace) && customToolNamespace is { Length: > 0 }
+				? customToolNamespace
+				: null;
+
+		ClassName = Utilities.GetClassNameFromPath(resxFilePath);
+
+		NullForgivingOperators = globalOptions.NullForgivingOperators;
+
+		PublicClass =
+			options.TryGetValue("build_metadata.EmbeddedResource.PublicClass", out var perFilePublicClassSwitch) &&
+			perFilePublicClassSwitch is { Length: > 0 }
+				? perFilePublicClassSwitch.Equals("true", StringComparison.OrdinalIgnoreCase)
+				: globalOptions.PublicClass;
+
+		StaticClass =
+			options.TryGetValue("build_metadata.EmbeddedResource.StaticClass", out var perFileStaticClassSwitch) &&
+			perFileStaticClassSwitch is { Length: > 0 }
+				? !perFileStaticClassSwitch.Equals("false", StringComparison.OrdinalIgnoreCase)
+				: globalOptions.StaticClass;
+
+		StaticMembers =
+			options.TryGetValue("build_metadata.EmbeddedResource.StaticMembers", out var staticMembersSwitch) &&
+			staticMembersSwitch is { Length: > 0 }
+				? staticMembersSwitch.Equals("false", StringComparison.OrdinalIgnoreCase)
```

Comment:
This line needs to be negated just like above, otherwise it generates non-static members when I set this option to `true`. It would be nice if you could write unit tests for options classes.

Reply:
Indeed, fixed